File: Lab5/Student.cpp
#include "Student.h"

Student::Student()
{
    this->fname = "Noname";
    this->date = "01.01.1901";
}

Student::Student(std::string fname, std::string date, int grpnum, float avratsc)
{ 
    this->fname = fname;
    this->date = date;
    this->grpnum = grpnum;
    this->avratsc = avratsc;
    if (avratsc > 95) this->scholstat = "Підвищена";
    else if(avratsc > 85) this->scholstat = "Звичайна";
    else this->scholstat = "Немає";
}   

Student::Student(std::string str)
{ 
    std::string fname, date;
    int grpnum;
    float avratsc;
    std::vector<std::string> words;
    int n1 = 0, n2 = str.find(';');
    for(int i=0; i<4; i++)
    {
        words.push_back(str.substr(n1,n2-n1));
        n1 = n2+1;
        n2 = str.find(';', n1+1);
    }
    words.push_back(str.substr(n1,str.size()-n1));
    fname = words[0];
    date = words[1];
    grpnum = stoi(words[2]);
    avratsc = stoi(words[3]);
    this->fname = fname;
    this->date = date;
    this->grpnum = grpnum;
    this->avratsc = avratsc;
    if (avratsc > 95) this->scholstat = "Підвищена";
    else if(avratsc > 85) this->scholstat = "Звичайна";
    else this->scholstat = "Немає";
} 

float Student::earn()
{
    std::string stat = this->scholstat;
    if (stat == "Підвищена") return 2000;
    else if(stat == "Звичайна") return 1000;
    else return 0;
}

int Student::GetGrnum()
{
    return this->grpnum;
}

float Student::GetAvrat()
{
    return this->avratsc;
}

std::string Student::GetStat()
{
    return this->scholstat;
}

File: Lab5/Python/Student.py
from Person import *
class Student(Person):
    def __init__(self, string):
        lst = string.split(";")
        super().__init__(lst[0], lst[1])
        self.__grpnum = int(lst[2]) # Номер групи
        self.__avratsc = int(lst[3]) # Середній рейтинговий бал
        if self.__avratsc > 95: self.__scholstat = "Підвищена"
        elif self.__avratsc > 85: self.__scholstat = "Звичайна"
        else: self.__scholstat = "Немає"

    # Підрахунок стипендії
    def earn(self):
        stat = self.__scholstat
        if stat == "Підвищена": return 2000
        elif stat == "Звичайна": return 1000
        else: return 0

    # Повертає номер групи
    def GetGrnum(self):
        return self.__grpnum

    # Повертає Середній рейтинговий бал
    def GetAvrat(self):
        return self.__avratsc

    # Повертає статус отримання стипендії    
    def GetStat(self):
        return self.__scholstat

    # Перевантаження виведення
    def __str__(self):
        return f"{self._name};{self._date};{self.__grpnum};{self.__avratsc}"